crates/rpc/eth/block.rs

```diff
@@ -400,7 +400,13 @@ pub fn get_transaction_by_hash(
             Some(transaction) => transaction,
             _ => return Ok(Value::Null),
         };
-
+    let (block_number, index) = storage
+        .get_transaction_location(request.transaction_hash)
+        .unwrap()
+        .unwrap();
+    let block_header = storage.get_block_header(block_number).unwrap().unwrap();
```

Comment:
Same as before... we could improve this instead of using `.unwrap().unwrap()`

Reply:
Sorry about this. I actually meant to create a draft PR. I will clean it up and mark it as ready for review asap.